src/Dnn.PersonaBar.Library/DTO/UserSettings.cs

```diff
@@ -20,41 +20,132 @@
 #endregion
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
 using System.Runtime.Serialization;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using DotNetNuke.Common.Utilities;
 
 namespace Dnn.PersonaBar.Library.DTO
 {
     /// <summary>
-    /// Persona Bar User Settings
+    /// Persona Bar Settings For User.
     /// </summary>
     [DataContract]
-    public class UserSettings
+    public class UserSettings : Dictionary<string, object>, IXmlSerializable
     {
-        [DataMember(Name = "expandPersonaBar")]
-        public bool ExpandPersonaBar{ get; set; }
+        [IgnoreDataMember]
+        public bool ExpandPersonaBar{
+            get => Convert.ToBoolean(this["expandPersonaBar"]);
+            set => this["expandPersonaBar"] = value;
+        }
 
-        [DataMember(Name = "activePath")]
-        public string ActivePath { get; set; }
+        public XmlSchema GetSchema()
+        {
+            return null;
+        }
 
-        [DataMember(Name = "activeIdentifier")]
-        public string ActiveIdentifier { get; set; }
+        public void ReadXml(XmlReader reader)
+        {
+            bool wasEmpty = reader.IsEmptyElement;
+            reader.Read();
 
-        [DataMember(Name = "expandTasksPane")]
-        public bool ExpandTasksPane { get; set; }
+            if (wasEmpty)
+                return;
 
-        [DataMember(Name = "comparativeTerm")]
-        public string ComparativeTerm { get; set; }
+            switch (reader.Name)
+            {
+                case "data":
+                    ReadSettings(reader);
+                    break;
+                default:
+                    //the method add for backward compatible, should remove this in future release.
```

Comment:
A simple comment will be easy to miss and no one will ever remove this method. Let's put this comment in an `[Obsolete]` attribute, saying what it says, so that we don't miss it in the future.